File: session-1/src/Sqrt.java
import java.util.Scanner;
public class Sqrt {
    public int mySqrt(int x) {
        if(x == 0)return 0;

        int left = 1, right = 46340;
        int mid = -1;
        int ans = -1;

        while(left <= right){
            mid = (left + right)/2;

            int mul = mid * mid;
            
            if(mul == x){
                ans = mid;
                break;
            }
            else if(mul < x){
                ans = Math.max(ans, mid);
                left = mid + 1;
            }else{
                right = mid - 1;
            }
        }
        return ans;
    }
    public static void main(String[] args) {
        Scanner input = new Scanner(System.in);
        Sqrt MySqrt = new Sqrt();

        int x = input.nextInt();
        System.out.println(MySqrt.mySqrt(x));
    }
}

File: session-1/src/Sqrt.py
class Solution:
    def mySqrt(self, x: int) -> int:
        if x == 0:
            return 0

        low = 1
        high = 46340

        answer = 0

        while low <= high:
            mid = (low + high) // 2

            if mid * mid == x:
                answer = mid
                break
            elif mid * mid < x:
                answer = max(answer, mid)
                low = mid + 1
            else:
                high = mid - 1

        return answer
